igor_src/main.cpp

```diff
@@ -1628,7 +1639,15 @@ int main(int argc , char* argv[]){
 				Aligner j_aligner (j_subst_matrix , j_gap_penalty , J_gene);
 				j_aligner.set_genomic_sequences(j_genomic);
 				try{
-					j_aligner.align_seqs(cl_path + "aligns/" +  batchname + j_align_filename , indexed_seqlist, j_align_thresh_value , j_best_only , j_left_offset_bound , j_right_offset_bound);
+					if (!gen_output_CDR3_data)
+						j_aligner.align_seqs(cl_path + "aligns/" +  batchname + j_align_filename , indexed_seqlist, j_align_thresh_value , j_best_only , j_left_offset_bound , j_right_offset_bound);
+					else{ //assume seqs are CDR3s
+						cout<<"running on CDRsonly"<<endl;
+						unordered_map<string,pair<int,int>> j_genomic_offset_bounds;
+						for(unordered_map<string,size_t>::const_iterator iter = j_CDR3_anchors.begin() ; iter != j_CDR3_anchors.end() ; iter++)
+							j_genomic_offset_bounds.emplace((*iter).first,make_pair( (*iter).second, (*iter).second )); 
```

Comment:
And your missing +/-3 factor comes from here
The anchor index indicates the first letter of the anchor codon, so in order to account for the full codon in the sequence you'll have to add +3

Reply:
My mistake, from the definition you've taken for the inverse offset 
`min_offset+=(rev_offset_frame)? seqlen:0;`
I think what you should put in for J is `-(*iter).second -2`